app/autodiscovery/thunderbird/src/test/java/com/fsck/k9/autodiscovery/thunderbird/ThunderbirdAutoconfigFetcherTest.kt

```diff
@@ -0,0 +1,44 @@
+package com.fsck.k9.autodiscovery.thunderbird
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import kotlin.test.assertNotNull
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Test
+
+class ThunderbirdAutoconfigFetcherTest {
+    private val fetcher = ThunderbirdAutoconfigFetcher(OkHttpClient.Builder().build())
+
+    @Test
+    fun shouldHandleNonexistentUrl() {
+        val nonExistentUrl =
+            "https://autoconfig.domain.invalid/mail/config-v1.1.xml?emailaddress=test%40domain.example".toHttpUrl()
+
+        val inputStream = fetcher.fetchAutoconfigFile(nonExistentUrl)
+
+        assertThat(inputStream).isNull()
+    }
+
+    @Test
+    fun shouldHandleEmptyResponse() {
+        val server = MockWebServer().apply {
+            this.enqueue(
+                MockResponse()
+                    .setBody("")
+                    .setResponseCode(204),
+            )
+            start()
+        }
+        val url = server.url("/empty/")
```

Comment:
I did see that there was a `MockWebServerHelper` that creates a mock server in a nicer way.
It is however defined in `com.fsck.k9.backend.jmap` and I wasn't sure if I should do all the work to include it in this project.

Reply:
This is fine. We can think about moving `MockWebServerHelper` to a common place when `MockWebServer` is used more often.